framework/ldtest/junit_test_logger.go

```diff
@@ -0,0 +1,206 @@
+package ldtest
+
+import (
+	"encoding/xml"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/launchdarkly/sdk-test-harness/framework"
+	"github.com/launchdarkly/sdk-test-harness/framework/harness"
+	"gopkg.in/launchdarkly/go-sdk-common.v2/ldvalue"
+)
+
+type JUnitTestLogger struct {
+	filePath    string
+	serviceInfo harness.TestServiceInfo
+	filters     RegexFilters
+	testIDs     []TestID // this slice preserves the order that the tests were run in
+	tests       map[string]jUnitTestStatus
+	lock        sync.Mutex
+}
+
+type jUnitTestStatus struct {
+	failures  []error
+	skipped   ldvalue.OptionalString
+	output    string
+	startTime time.Time
+	duration  time.Duration
+}
+
+// Struct definitions for the JUnit XML schema - see https://github.com/jstemmer/go-junit-report
+
+type jUnitXMLDocument struct {
+	XMLName xml.Name            `xml:"testsuites"`
+	Suites  []jUnitXMLTestSuite `xml:"testsuite"`
+}
+
+type jUnitXMLTestSuite struct {
+	XMLName    xml.Name           `xml:"testsuite"`
+	Tests      int                `xml:"tests,attr"`
+	Failures   int                `xml:"failures,attr"`
+	Time       string             `xml:"time,attr"`
+	Name       string             `xml:"name,attr"`
+	Properties []jUnitXMLProperty `xml:"properties>property,omitempty"`
+	TestCases  []jUnitXMLTestCase `xml:"testcase"`
+}
+
+type jUnitXMLTestCase struct {
+	XMLName     xml.Name             `xml:"testcase"`
+	Classname   string               `xml:"classname,attr"`
```

Comment:
There isn't really a "class" associated with any of these tests, so currently I'm leaving this field blank, which I think CircleCI will be OK with. The `go-junit-report` tool normally sets this to the _package_ name, but there isn't really a package for this either other than "sdktests".